internal/metrics/disk_stats_helper.go

```diff
@@ -22,12 +24,22 @@ func NewOSDiskStatsHelper() *OSDiskStatsHelper {
 
 func (c *OSDiskStatsHelper) getUsedBytesInDirectory(directory string) (uint64, error) {
 	var totalBytes uint64
+	// Maintain a map of inodes we have seen so that we don't double count storage
+	seenInodes := make(map[uint64]bool)
 	err := filepath.Walk(directory, func(path string, fileInfo os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-		if !fileInfo.IsDir() {
+		// We need to ensure that the size check is based on inodes and not just the sizes gleaned from files.
+		// we should ensure that we have not seen a particular inode for a given file before.
+		// this is because there are hard links in use and we need to account for this.
+		stat, ok := fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
Wouldn't it be simpler by using `syscall` package? Is there any issue with that solution?

https://pkg.go.dev/syscall#Statfs

Reply:
Nevermind it seems it doesn't do what I thought it would